scrapper.py

```diff
@@ -17,47 +34,152 @@ class NumberOfArticlesOutOfRangeError(Exception):
 
 class IncorrectNumberOfArticlesError(Exception):
     """
-    Total number of articles to parse in not integer
+    Total number of articles to parse is not integer  # in|is !typo
+    """
+
+
+class BadStatusCode(Exception):
+    """
+    Request did not give valid response
     """
 
 
 class Crawler:
     """
     Crawler implementation
     """
-    def __init__(self, seed_urls, max_articles: int):
-        pass
+
+    def __init__(self, seed_urls: List[str], max_articles: int):
+        self.seed_urls = seed_urls
+        self.total_max_articles = max_articles
+        self.urls = []
 
     def _extract_url(self, article_bs):
-        pass
+        contents_bs = article_bs.select('div.content-cont-col-text a')
+        links = ['https://alp.iling.spb.ru/' + content['href'][2:] for content in contents_bs]
+        self.urls.extend(links)
 
     def find_articles(self):
         """
         Finds articles
         """
-        pass
+        for url in self.seed_urls:
+            page = obtain_page(url)
+            soup = BeautifulSoup(page, 'lxml')
+            self._extract_url(soup)
+            sleep(random.uniform(2, 4))
+            if len(self.urls) < self.total_max_articles:
+                continue
+            self.urls = self.urls[:self.total_max_articles]
```

Comment:
you may avoid using continue if you will check for NOT condition

Reply:
Fixed.